### ISSUE TITLE
nodejsscan exception: "No such file or directory"

### ISSUE THREAD

**Issue by hoop33** [AppThreat/sast-scan#24]
# TL;DR

`sast-scan` passes a full path to `nodejsscan` for the output, and `nodejsscan` can't handle it. `nodejsscan` run outside of `sast-scan` handles the same parameters fine. Does the Docker image have a different version of `nodejsscan`?

# Problem

I'm using sast-scan in GitLab CICD to scan a node project. I've tried a few different parameters, but no matter what I specify for `out_dir` I get a "No such file or directory" error. Here's what I have in my `.gitlab-ci.yml` file:

```yml
sast:
  stage: test
  image: 
    name: appthreat/sast-scan
  script:
    - scan --src src --type nodejs --out_dir .
  artifacts:
    paths:
      - nodejsscan-report.json
```
And here's the output:

```
$ scan --src src --type nodejs --out_dir .
  ___            _____ _                    _
 / _ \          |_   _| |                  | |
/ /_\ \_ __  _ __ | | | |__  _ __ ___  __ _| |_
|  _  | '_ \| '_ \| | | '_ \| '__/ _ \/ _` | __|
| | | | |_) | |_) | | | | | | | |  __/ (_| | |_
\_| |_/ .__/| .__/\_/ |_| |_|_|  \___|\__,_|\__|
      | |   | |
      |_|   |_|
INFO [2020-04-10 12:53:54,342] Scanning src using scan plugins ['nodejs']
INFO [2020-04-10 12:53:54,342] ================================================================================
INFO [2020-04-10 12:53:54,343] ⚡︎ Executing "nodejsscan --output ./nodejsscan-report.json -d src"
ERROR [2020-04-10 12:53:54,610] Received exception: [Errno 2] No such file or directory: './nodejsscan-report.json'
```
# What I've Tried

I've tried different directories, and get the same results. As soon as it's a path to a directory, `nodesjsscan` shows that error.

I've tried not specifying `--out_dir`, but then `sast-scan` constructs a full path `src/reports/nodejsscan-report.json` that it passes to `nodejsscan` and I get the same error.

I tried bypassing `sast-scan` entirely and calling `nodejsscan` directly so I could specify the parameters directly:

```yml
sast:
  stage: test
  image: 
    name: opensecurity/nodejsscan:cli
    entrypoint:
      - "/usr/bin/env"
      - "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
  script:
    - python /usr/src/app/cli.py -d src -o sast-report.json
  artifacts:
    paths:
      - sast-report.json
```
That works. Note that there's no path in the output parameter.

I tried running `nodejsscan` locally with the parameters that `sast-scan` passes:

```sh
$ nodejsscan --output ./nodejsscan-report.json -d src
```
That works.

# Where I'm Confused

The `sast-scan` image on docker hub was built 3 days ago. The version of `nodejsscan` on PyPI is `3.7`, same as I'm running locally, released Jan 18, 2020. I'm looking at the code where the error seems to be thrown, and that hasn't changed in over a year.

**Comment by prabhu** [comment]
This is a great report! Really, appreciate the amount of effort you have put in and the kind of pain points you have gone through because of a lack of documentation for GitLab CI integration on this repo.

I just used the environment variable CI_PROJECT_DIR exposed by GitLab CI.

```yaml
sast:
  stage: test
  image: 
    name: appthreat/sast-scan
  script:
    - scan --src ${CI_PROJECT_DIR} --type nodejs --out_dir ${CI_PROJECT_DIR}/reports
  artifacts:
    paths:
      - $CI_PROJECT_DIR/reports/nodejsscan-report.json
```

https://gitlab.com/prabhu3/NodeGoat/-/jobs/506830325

Feel free to upload the whole reports directory which will also include the .sarif file along with .json file.

I will add this configuration to the docs folder as well.

**Comment by hoop33** [comment]
Thanks for that -- that worked! Closing issue.
